### PR TITLE
ICMSLST-2034 Access Approval Request emails

### PR DESCRIPTION
**Pull request by marcuspp**
Sends Access Approval emails when an Approval is created and when a response has been received.
Also updated the `Responses` and `Statuses` on the model to be TypedTextChoices

### CODE REVIEW

File: web/domains/case/access/approval/models.py [hunk 1]
```diff
@@ -15,22 +16,15 @@ class Meta:
...
-        (DRAFT, "DRAFT"),
-        (OPEN, "OPEN"),
-        (CANCELLED, "CANCELLED"),
-        (COMPLETED, "COMPLETED"),
-    )
+    class Statuses(TypedTextChoices):
+        DRAFT = ("DRAFT", "DRAFT")
+        OPEN = ("OPEN", "OPEN")
+        CANCELLED = ("CANCELLED", "CANCELLED")
+        COMPLETED = ("COMPLETED", "COMPLETED")
```
Comment (MattHolmes123) [inline web/domains/case/access/approval/models.py:27]: I literally had an open todo in my local repo to move these values to where the ImpExpStatus lives:
image

File: web/domains/case/access/approval/models.py [hunk 2]
```diff
@@ -15,22 +16,15 @@ class Meta:
...
-    OPEN = "OPEN"
-    CANCELLED = "CANCELLED"
-    COMPLETED = "COMPLETED"
-    STATUSES = (
-        (DRAFT, "DRAFT"),
-        (OPEN, "OPEN"),
-        (CANCELLED, "CANCELLED"),
-        (COMPLETED, "COMPLETED"),
-    )
+    class Statuses(TypedTextChoices):
```
Comment (MattHolmes123) [inline web/domains/case/access/approval/models.py:23]: I prefer singular to fit with `ImpExpStatus` so `Status`

Comment (MattHolmes123) [reply web/domains/case/access/approval/models.py:23]:
```
class ImportApplicationType(models.Model):
    class Types(TypedTextChoices):
        DEROGATION = ("SAN", "Derogation from Sanctions Import Ban")
        FIREARMS = ("FA", "Firearms and Ammunition")  # has subtypes
        IRON_STEEL = ("IS", "Iron and Steel (Quota)")
        OPT = ("OPT", "Outward Processing Trade")
        SANCTION_ADHOC = ("ADHOC", "Sanctions and Adhoc Licence Application")
        SPS = ("SPS", "Prior Surveillance")
        TEXTILES = ("TEX", "Textiles (Quota)")
        WOOD_QUOTA = ("WD", "Wood (Quota)")

    class SubTypes(TypedTextChoices):
        OIL = ("OIL", "Open Individual Import Licence")
        DFL = ("DEACTIVATED", "Deactivated Firearms Import Licence")
        SIL = ("SIL", "Specific Individual Import Licence")
```
As I say that these are plural...

Comment (MattHolmes123) [reply web/domains/case/access/approval/models.py:23]: ok so there is a 50 / 50 split on TypedTextChoices on singular / plural so do whatever :D

Comment (MattHolmes123) [reply web/domains/case/access/approval/models.py:23]: singular saves characters.
If I had a wand I'd make them all singular :)

**Review by mikeylockwood — APPROVED** [review]

**Review by MattHolmes123 — APPROVED** [review]

**Pull request merged**